internal/pkg/ccloudv2/schema_registry.go

```diff
@@ -51,3 +64,51 @@ func (c *Client) ListSchemaRegistryRegions(cloud, packageType string) ([]srcm.Sr
 	}
 	return regionList, nil
 }
+
+func (c *Client) GetSchemaRegistryClusterById(clusterId, environment string) (srcmv2.SrcmV2Cluster, error) {
+	cluster, httpResp, err := c.SchemaRegistryClient.ClustersSrcmV2Api.GetSrcmV2Cluster(c.SchemaRegistryApiContext(), clusterId).Environment(environment).Execute()
+	return cluster, errors.CatchCCloudV2Error(err, httpResp)
+}
+
+func (c *Client) DeleteSchemaRegistryCluster(clusterId, environment string) error {
+	httpResp, err := c.SchemaRegistryClient.ClustersSrcmV2Api.DeleteSrcmV2Cluster(c.SchemaRegistryApiContext(), clusterId).Environment(environment).Execute()
+	return errors.CatchCCloudV2Error(err, httpResp)
+}
+
+func (c *Client) UpgradeSchemaRegistryCluster(srcmV2ClusterUpdate srcmv2.SrcmV2ClusterUpdate, clusterId string) (srcmv2.SrcmV2Cluster, error) {
+	cluster, httpResp, err := c.SchemaRegistryClient.ClustersSrcmV2Api.UpdateSrcmV2Cluster(c.SchemaRegistryApiContext(), clusterId).SrcmV2ClusterUpdate(srcmV2ClusterUpdate).Execute()
+	return cluster, errors.CatchCCloudV2Error(err, httpResp)
+}
+
+func (c *Client) GetSchemaRegistryClustersByEnvironment(environment string) ([]srcmv2.SrcmV2Cluster, error) {
+	var list []srcmv2.SrcmV2Cluster
+
+	done := false
+	pageToken := ""
+	for !done {
+		page, httpResp, err := c.executeListSchemaRegistryClusters(environment, pageToken)
+		if err != nil {
+			return nil, errors.CatchCCloudV2Error(err, httpResp)
+		}
+		list = append(list, page.GetData()...)
+
+		pageToken, done, err = extractNextPageToken(page.GetMetadata().Next)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if len(list) == 0 {
+		return nil, errors.NewSRNotEnabledError()
+	}
```

Comment:
This is a bit different than our normal style. Can we move this outside of this function?